src/components/Block/Block.styles.ts

```diff
@@ -15,19 +17,38 @@ const getMarginTop = (topSpacing?: GutterSize, push?: number) => {
   }
 };
 
+const iconSizeForTextSize = {
+  [TextSize.XXLARGE]: '2.4rem',
+  [TextSize.XLARGE]: '2.2rem',
+  [TextSize.LARGE]: '1.5rem',
+  [TextSize.MEDIUM]: '1.4rem',
+  [TextSize.MEDIUM_SUB]: '1.4rem',
+  [TextSize.SMALL]: '1.2rem',
+  [TextSize.XSMALL]: '1.0rem',
+};
+
 export const getStyles = (props: BlockProps): IRawStyle => {
   const { push, textSize, topSpacing, bottomSpacing, textAlign, textColor } = props;
+  const theme = getTheme();
```

Comment:
Todo in another PR: add memoization

Reply:
Should we try adding a `Customizer` in one of our consumer apps? I'm kind of excited to see this working.

Reply:
Yeah, me too. I'll look at this next.